eZ/Publish/Core/SignalSlot/RoleService.php

```diff
@@ -66,8 +67,243 @@ public function __construct(RoleServiceInterface $service, SignalDispatcher $sig
     }
 
     /**
+     * Creates a new RoleDraft.
+     *
+     * @since 6.0
+     *
+     * @throws \eZ\Publish\API\Repository\Exceptions\UnauthorizedException if the authenticated user is not allowed to create a role
+     * @throws \eZ\Publish\API\Repository\Exceptions\InvalidArgumentException if the name of the role already exists or if limitation of the
+     *                                                                        same type is repeated in the policy create struct or if
+     *                                                                        limitation is not allowed on module/function
+     * @throws \eZ\Publish\API\Repository\Exceptions\LimitationValidationException if a policy limitation in the $roleCreateStruct is not valid
+     *
+     * @param \eZ\Publish\API\Repository\Values\User\RoleCreateStruct $roleCreateStruct
+     *
+     * @return \eZ\Publish\API\Repository\Values\User\RoleDraft
+     */
+    public function createRoleDraft(RoleCreateStruct $roleCreateStruct)
+    {
+        $returnValue = $this->service->createRoleDraft($roleCreateStruct);
+        $this->signalDispatcher->emit(
+            //TODO: CreateRoleDraftSignal? Or skip it altogether?
```

Comment:
Think use-cases. Would signals in this area commonly need to distinguish drafts ?
Maybe an argument to the signal would be sufficient ?